### PR TITLE
kubeadm: add `kubeadm config print reset-defaults` command for ResetConfiguration

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
kubeadm: add `kubeadm config print reset-defaults` command for ResetConfiguration

#### Which issue(s) this PR fixes:

Part of: kubernetes/kubeadm#2890

#### Special notes for your reviewer:
Sample output：
```sh
$ kubeadm config print reset-defaults                                                
apiVersion: kubeadm.k8s.io/v1beta4
certificatesDir: /etc/kubernetes/pki
criSocket: unix:///var/run/containerd/containerd.sock
kind: ResetConfiguration
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by pacoxu** [comment]
/priority important-soon
/triage accepted
/lgtm
（late）

File: cmd/kubeadm/app/cmd/config_test.go [hunk 1]
```diff
@@ -451,6 +451,32 @@ func TestNewCmdConfigPrintActionDefaults(t *testing.T) {
...
+			name: "JoinConfiguration: KubeProxyConfiguration",
+			expectedKinds: []string{
+				"KubeProxyConfiguration",
+				constants.ResetConfigurationKind,
+			},
+			componentConfigs: "KubeProxyConfiguration",
+			cmdProc:          newCmdConfigPrintResetDefaults,
+		},
+		{
+			name: "JoinConfiguration: KubeProxyConfiguration and KubeletConfiguration",
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/config_test.go:471]: `ReseteConfiguration: KubeProxyConfiguration and KubeletConfiguration`

File: cmd/kubeadm/app/cmd/config_test.go [hunk 2]
```diff
@@ -451,6 +451,32 @@ func TestNewCmdConfigPrintActionDefaults(t *testing.T) {
...
 		},
+		{
+			name: "ResetConfiguration: No component configs",
+			expectedKinds: []string{
+				constants.ResetConfigurationKind,
+			},
+			cmdProc: newCmdConfigPrintResetDefaults,
+		},
+		{
+			name: "JoinConfiguration: KubeProxyConfiguration",
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/config_test.go:462]: ResetConfiguration: KubeProxyConfiguration

File: cmd/kubeadm/app/cmd/config_test.go (hunk 1 above)
Comment (chendave) [reply cmd/kubeadm/app/cmd/config_test.go:471]: and thinking this more, is there any case for the resetConfiguration to be configured along with component configs? I don't think so, we should just remove those cases.

**Review by chendave — COMMENTED** [review]
thanks @SataQiu !

**Comment by chendave** [comment]
propose to remove the testcases: https://github.com/kubernetes/kubernetes/pull/119346

**Pull request merged**